src/main/java/com/billyyccc/mssqlclient/impl/protocol/MessageStatus.java

```diff
@@ -0,0 +1,37 @@
+package com.billyyccc.mssqlclient.impl.protocol;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public enum MessageStatus {
+
+  NORMAL(0x00),
+  END_OF_MESSAGE(0x01),
+  IGNORE_THIS_EVENT(0x02),
+  RESET_CONNECTION(0x08),
+  RESET_CONNECTION_SKIP_TRAN(0x10);
+
+  private static final Map<Integer, MessageStatus> typeMapping = new HashMap<>();
```

Comment:
How about using an array instead ? it seems that the number of values is quite small and using an array lookup will be much faster